### PR TITLE
Details sheet animation

### PR DESCRIPTION
**Pull request by osdnk**
Fixes  TEAM2-283

Figma link (if any):

## Screen recordings / screenshots
https://user-images.githubusercontent.com/25709300/194029503-69c00070-6e10-42f6-934c-9e8aa0a80a6b.mov

Unfortunately, I can not make it ideal due to technical limitations. We can plan some long-term strategies, but realistically I cannot propose any "perfect" solution that will take us shorter than ~1 week on android or a few weeks on iOS.

Ideally, on iOS I would expose the sheet position as an animated value and then manipulate the button position using this. Unfortunately, cool modals do not allow for this. We can target upgrading our dependency on rn-screen (coz it's supported in the newer versions), but this is a massive effort and a ton of changes, which are probably not worth it at all. We have this speced-out already, but looks like there's no initiative to work on it,

On Android, here we are using the react-navigation sheet. This sheet is not giving us the animated position. This could have happened while migrating the sheet to the "bottom sheet". Also, looks like this is not that bad on android. Frankly, the animation there looks so rough itself that this is not proper prioritizing to focus on improving just the button at this point. 

Hence, I fixed this only on iOS in a slightly heuristic way. 
1. Firstly, while the screen is mounted and flex calculated, we migrate the button to the absolute position aligning to avoid flickering while making the following steps
2. When the button is pressed, we calculate the target position of the button
3. Then we animate the button to the target position in such a way that this is not jumping that much and it looks smooth.

Unfortunately, this is not giving a precisely "fixed" position and the button is still moving slightly. I cannot fix this perfectly, because there is also a few ms de-synchronizations. However, this looks better (see the video) and therefore, should be considered as an improvement. 

## What to test
Try the animation on iOS.

### CODE REVIEW

**Review by pugson — APPROVED** [review]
probably as good as it's gonna get without major changes you outlined. lgtm 👍️

**Review by estebanmino — APPROVED** [review]
awesome, code lgtm, ios working fine and android working as before https://www.loom.com/share/703c6c388eda4710bc25e602b3d295ad

File: src/components/expanded-state/useSwapDetailsButtonPosition.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    []
+  );
+
+  return {
+    wrapperStyle,
+    buttonWrapperStyle,
+    onHeightChange,
+    justOpenedSheet,
+    onPressMore,
+    onWrapperLayout,
```
Comment (estebanmino) [inline src/components/expanded-state/useSwapDetailsButtonPosition.ts:63]: actually `buttonWrapperStyle` and `justOpenedSheet` is not used where the hook is called, don't have string opinions about removing if we're not using so not blocker

Comment (osdnk) [reply src/components/expanded-state/useSwapDetailsButtonPosition.ts:63]: ah right, thanks!

**Review by skylarbarrera — APPROVED** [review]
👻 

seems marginally better so :shipit:

**Comment by estebanmino** [comment]
thank you @osdnk ! 🙏 🐐

**Pull request merged**